builtins/match.go

```diff
@@ -57,7 +66,7 @@ func matchRecordOf(val, pat sliceHolder) (bool, error) {
 	i, back_i := 0, -1
 	j, back_j := 0, -1
 	for i < val.Len() && j < pat.Len() {
-		if pat.Get(j) == runtime.AnyOrNone {
+		if pat.Get(j) == runtime.AnyOrNone || pat.Get(j).Equal(runtime.NewString("*")) {
```

Comment:
This is also a good place for a explanatory comment, why this Equal call is required.